modules/tag-input/main/index.coffee

```diff
@@ -87,19 +88,22 @@ class TagInput extends hx.EventEmitter
         if @input.value() is ''
           selection = @tagContainer.selectAll('.hx-tag')
           if selection.size() > 0
+            @_.autocomplete?.hide()
             nodeSelection = hx.select(selection.node(selection.size()-1))
             value = nodeSelection.text()
             nodeSelection.remove()
+            @_.autocomplete?.show()
             @emit 'remove', {value: value, type: 'user'}
 
     @input.on 'keyup', 'hx.tag-input', (event) ->
       if (event.keyCode or event.charCode) is 8
         backspacedown = false
         true
 
-    @input.on 'blur', 'hx.tag-input', (event) =>
-      if @input.value().length > 0 and not hasError()
-        @add(@input.value(), undefined)
+    if not @_.autocomplete
+      @input.on 'blur', 'hx.tag-input', (event) =>
```

Comment:
We don't need the blur event at all when we have an autocomplete as `AutoComplete` deals with it